web/packages/teleport/src/Roles/RoleEditor/standardmodel.ts

```diff
@@ -206,6 +244,13 @@ export type WindowsDesktopAccessSpec = AccessSpecBase<'windows_desktop'> & {
   logins: readonly Option[];
 };
 
+export type RuleModel = {
+  /** Autogenerated ID to be used with the `key` property. */
+  id: string;
+  resources: readonly ResourceKindOption[];
+  verbs: readonly VerbOption[];
```

Comment:
I think the model should be kept as lean as possible; values that can be derived shouldn't be part of state https://react.dev/learn/choosing-the-state-structure#avoid-redundant-state.

Are there any significant drawbacks of the current approach? I’d say one potential downside is the inconvenience of accessing values like `verb.value` instead of just `verb` at higher levels in the component tree, that don't care about the labels.

But I think we can move forward with the current implementation. I’m aware that `ReactSelect` can be inconvenient to work with when using plain values, as it only accepts objects. Also I'm not sure if this is worth spending time on it at this point.